jax/_src/numpy/fft.py

```diff
@@ -62,6 +60,12 @@ def _fft_core(func_name, fft_type, a, s, axes, norm):
         s += [max(0, 2 * (a.shape[axes[-1]] - 1))]
     else:
       s = [a.shape[axis] for axis in axes]
+  else:
+    if fft_type == xla_client.FftType.IRFFT:
+      _s = tuple(list(s)[:-1] + [max(0, s[-1] // 2 + 1)])
+      a = _fix_shape(a, _s, axes)
+    else:
+      a = _fix_shape(a, s, axes)
```

Comment:
If you get the chance, maybe take a look at https://github.com/google/jax/pull/6266, which is also trying to add support for setting `s` in FFT?

Ideally we would merge this change as a separate PR, first.

Reply:
Yes, https://github.com/google/jax/pull/6266 seems to do largely the same as commit https://github.com/google/jax/pull/6343/commits/9291632ea331a4b149ebd6e798505b4c50b64a67 from this PR.

Differences from quickly reading through the code:
-  Explicit use of padding (theirs), vs copying in an array of zeros (mine)
- Only pad when needed (mine), vs pad when a change is needed (theirs). (Maybe `lax.pad` can actually decide not to do any padding operation when it notices this is not needed, I'm not familiar with these optimizations).
- Separate tests with and without shape (mine).
- I think their tests don't test combinations of both crop and pad (From quickly reading through the code, I might be wrong.)

In general I think they are roughly the same, and I can update this PR once https://github.com/google/jax/pull/6266 has been merged.